scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala

```diff
@@ -390,6 +417,42 @@ class BigQueryClient private (private val projectId: String,
   }
 
   // =======================================================================
+  // Legacy vs SQL 2011
+  // =======================================================================
+
+  private val isLegacySqlCache: LoadingCache[(String, Boolean), JBoolean] = CacheBuilder
+    .newBuilder()
+    .build(new CacheLoader[(String, Boolean), JBoolean] {
+      override def load(key: (String, Boolean)): JBoolean =
+        new JBoolean(isLegacySqlImpl(key._1, key._2))
+    })
+
+  private def isLegacySql(sqlQuery: String, flattenResults: Boolean): Boolean =
+    isLegacySqlCache.get((sqlQuery, flattenResults))
+
+  private def isLegacySqlImpl(sqlQuery: String, flattenResults: Boolean): Boolean = {
+    def run(useLegacySql: Boolean) = {
+      val queryConfig = createJobConfigurationQuery(
+        sqlQuery, temporaryTable, flattenResults, useLegacySql)
+      val jobConfig = new JobConfiguration().setQuery(queryConfig).setDryRun(true)
+      val job = new Job().setConfiguration(jobConfig)
+      bigquery.jobs().insert(projectId, job).execute()
+    }
+    try {
+      run(false)
+      false
+    } catch {
+      case e: GoogleJsonResponseException =>
+        if (e.getDetails.getErrors.get(0).getReason == "invalidQuery") {
```

Comment:
That class doesn't have a "invalidQuery" reason and `getErrorInfo` is protected :(